pkg/admission/admissiontimeout/timeoutadmission.go

```diff
@@ -0,0 +1,63 @@
+package admissiontimeout
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"k8s.io/apiserver/pkg/admission"
+)
+
+type pluginHandlerWithTimeout struct {
+	name            string
+	admissionPlugin admission.Interface
+	timeout         time.Duration
+}
+
+var _ admission.ValidationInterface = &pluginHandlerWithTimeout{}
+var _ admission.MutationInterface = &pluginHandlerWithTimeout{}
+
+func (p pluginHandlerWithTimeout) Handles(operation admission.Operation) bool {
+	return p.admissionPlugin.Handles(operation)
+}
+
+func (p pluginHandlerWithTimeout) Admit(a admission.Attributes, o admission.ObjectInterfaces) error {
+	mutatingHandler, ok := p.admissionPlugin.(admission.MutationInterface)
+	if !ok {
+		return nil
+	}
+
+	admissionDone := make(chan struct{})
+	var admissionErr error
+	go func() {
+		admissionErr = mutatingHandler.Admit(a, o)
```

Comment:
do we want to add a guard e.g `defer runtime.HandleCrash()`?

Reply:
> do we want to add a guard e.g `defer runtime.HandleCrash()`?

yes, good catch.